src/main/java/com/gtnewhorizons/angelica/glsm/debug/OpenGLDebugging.java

```diff
@@ -0,0 +1,441 @@
+        new GLproperty(GL11.GL_UNPACK_ALIGNMENT, "GL_UNPACK_ALIGNMENT", "Value of GL_UNPACK_ALIGNMENT", "pixel-store", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_UNPACK_LSB_FIRST, "GL_UNPACK_LSB_FIRST", "Value of GL_UNPACK_LSB_FIRST", "pixel-store", GL_GET_BOOLEANV),
+        new GLproperty(GL11.GL_UNPACK_ROW_LENGTH, "GL_UNPACK_ROW_LENGTH", "Value of GL_UNPACK_ROW_LENGTH", "pixel-store", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_UNPACK_SKIP_PIXELS, "GL_UNPACK_SKIP_PIXELS", "Value of GL_UNPACK_SKIP_PIXELS", "pixel-store", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_UNPACK_SKIP_ROWS, "GL_UNPACK_SKIP_ROWS", "Value of GL_UNPACK_SKIP_ROWS", "pixel-store", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_UNPACK_SWAP_BYTES, "GL_UNPACK_SWAP_BYTES", "Value of GL_UNPACK_SWAP_BYTES", "pixel-store", GL_GET_BOOLEANV),
+        new GLproperty(GL11.GL_VERTEX_ARRAY, "GL_VERTEX_ARRAY", "Vertex array enable", "vertex-array", GL_IS_ENABLED),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_POINTER, "GL_VERTEX_ARRAY_POINTER", "Pointer to the vertex array", "vertex-array", "glGetPointerv()"),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_SIZE, "GL_VERTEX_ARRAY_SIZE", "Coordinates per vertex", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_STRIDE, "GL_VERTEX_ARRAY_STRIDE", "Stride between vertices", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_TYPE, "GL_VERTEX_ARRAY_TYPE", "Type of vertex coordinates", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VIEWPORT, "GL_VIEWPORT", "Viewport origin and extent", "viewport", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_ZOOM_X, "GL_ZOOM_X", "x zoom factor", "pixel", GL_GET_FLOATV),
+        new GLproperty(GL11.GL_ZOOM_Y, "GL_ZOOM_Y", "y zoom factor", "pixel", GL_GET_FLOATV),
+    };
+
+    public static void dumpToFile(Consumer<Consumer<String>> dumper, String fileName) {
+        final List<String> lines = new ArrayList<>();
+        dumper.accept(lines::add);
+        try {
+            FileUtils.writeLines(new File(fileName), lines);
+        } catch (IOException e) {
+            LOGGER.error("Failed to write file");
+            e.printStackTrace();
+        }
+    }
+
+    public static void dumpStateToFile() {
+        dumpToFile((p) -> dumpState(p, true), "gl-dump-state.txt");
+    }
+    public static void dumpGLSMStateToFile() {
+        dumpToFile((p) -> dumpState(p, true), "glsm-dump-state.txt");
+    }
+
+    public static void dumpState(boolean glsm) {
+        dumpState(LINE_LOGGER, glsm);
+    }
+
+    public static void dumpState(Consumer<String> printer, boolean glsm) {
+        for (int i = 0; i < instance.propertyList.length; ++i) {
+            final GLproperty gLproperty = instance.propertyList[i];
+            printer.accept(gLproperty.name + ":" + gLproperty.getAsString(glsm) + " (" + gLproperty.description + ")");
+        }
+    }
+
+    public static void dumpIsEnabledToFile() {
```

Comment:
No `glsm` flag for this one? Could probably be done using `instance.propertyList[i].getAsString(glsm)`